src/main/java/com/metamx/http/client/response/SequenceInputStreamResponseHandler.java

```diff
@@ -96,14 +97,12 @@ public InputStream nextElement()
 
   @Override
   public ClientResponse<InputStream> handleChunk(
-      ClientResponse<InputStream> clientResponse, HttpChunk chunk
+      ClientResponse<InputStream> clientResponse, HttpContent chunk
   )
   {
-    final ChannelBuffer channelBuffer = chunk.getContent();
-    final int bytes = channelBuffer.readableBytes();
-    if (bytes > 0) {
+    if (chunk.content().hasArray()) {
```

Comment:
hasArray() means not that there are no bytes, it means that the ByteBuf is backed by off-heap memory.